bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java

```diff
@@ -1,314 +1,337 @@
+        public int hashCode() {
+            return Objects.hash(chatId, replyId);
+        }
+
+        @Override
+        public boolean equals(@Nullable Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            ReplyKey other = (ReplyKey) obj;
+            return Objects.equals(chatId, other.chatId) && Objects.equals(replyId, other.replyId);
+        }
+    }
+
+    private final List<Long> chatIds = new ArrayList<Long>();
+    private final Logger logger = LoggerFactory.getLogger(TelegramHandler.class);
+    private @Nullable ScheduledFuture<?> thingOnlineStatusJob;
+
+    // Keep track of the callback id created by Telegram. This must be sent back in
+    // the answerCallbackQuery
+    // to stop the progress bar in the Telegram client
+    private final Map<ReplyKey, String> replyIdToCallbackId = new HashMap<>();
+    // Keep track of message id sent with reply markup because we want to remove the
+    // markup after the user provided an
+    // answer and need the id of the original message
+    private final Map<ReplyKey, Integer> replyIdToMessageId = new HashMap<>();
+
+    private @Nullable TelegramBot bot;
+    private @Nullable OkHttpClient botLibClient;
+    private @Nullable HttpClient downloadDataClient;
+    private @Nullable ParseMode parseMode;
+
+    public TelegramHandler(Thing thing, @Nullable HttpClient httpClient) {
+        super(thing);
+        downloadDataClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands to handle
+    }
+
+    @Override
+    public void initialize() {
+        TelegramConfiguration config = getConfigAs(TelegramConfiguration.class);
+
+        String botToken = config.getBotToken();
+        chatIds.clear();
+        for (String chatIdStr : config.getChatIds()) {
+            try {
+                chatIds.add(Long.valueOf(chatIdStr));
+            } catch (NumberFormatException e) {
+                logger.warn("The chat id {} is not a number and will be ignored", chatIdStr);
+            }
+        }
+        if (config.getParseMode() != null) {
+            try {
+                parseMode = ParseMode.valueOf(config.getParseMode());
+            } catch (IllegalArgumentException e) {
+                logger.warn("parseMode is invalid and will be ignored. Only Markdown or HTML are allowed values");
+            }
+        }
+
+        OkHttpClient.Builder prepareConnection = new OkHttpClient.Builder().connectTimeout(75, TimeUnit.SECONDS)
+                .readTimeout(75, TimeUnit.SECONDS);
+
+        String proxyHost = config.getProxyHost();
+        Integer proxyPort = config.getProxyPort();
+        String proxyType = config.getProxyType();
+
+        if (proxyHost != null) {
```

Comment:
```suggestion
        if (proxyHost != null && proxyPort !=null) {
```

there is no guarantee that `proxyPort` is non-null when `proxyHost` is non-null. If it is null you'll get a NPE in l. 159.